apps/server/src/modules/files-storage/uc/files-storage.uc.spec.ts

```diff
@@ -207,148 +178,225 @@ describe('FilesStorageUC', () => {
-						status: 'pending',
-						updatedAt: expect.any(Date) as Date,
-					},
-					size: 10699,
-					updatedAt: expect.any(Date) as Date,
-				})
-			);
+		describe('WHEN user is not authorised', () => {
+			const setup = () => {
+				const { userId, uploadFromUrlParams } = getUploadFromUrlParams();
+				const error = new Error('test');
+				authorizationService.checkPermissionByReferences.mockRejectedValueOnce(error);
+
+				return { uploadFromUrlParams, userId, error };
+			};
+
+			it('should pass error', async () => {
+				const { uploadFromUrlParams, userId, error } = setup();
+
+				await expect(filesStorageUC.uploadFromUrl(userId, uploadFromUrlParams)).rejects.toThrow(error);
+			});
 		});
 
-		it('should throw Error', async () => {
-			httpService.get.mockResolvedValue(
-				createObservable({
+		describe('WHEN httpService throws error', () => {
+			const setup = () => {
+				const { userId, uploadFromUrlParams } = getUploadFromUrlParams();
+
+				const error = createObservable({
 					isAxiosError: true,
 					code: '404',
 					response: {},
 					name: 'errorText',
 					message: 'errorText',
 					toJSON: () => ({}),
-				}) as never
-			);
+				}) as never;
+				httpService.get.mockResolvedValue(error);
 
-			await expect(filesStorageUC.uploadFromUrl(userId, uploadFromUrlParams)).rejects.toThrow(NotFoundException);
-		});
+				return { uploadFromUrlParams, userId };
+			};
 
-		describe('Tests of permission handling', () => {
-			it('should call authorizationService.hasPermissionByReferences', async () => {
-				authorizationService.checkPermissionByReferences.mockResolvedValue();
-				await filesStorageUC.uploadFromUrl(userId, uploadFromUrlParams);
-				expect(authorizationService.checkPermissionByReferences).toBeCalledWith(
-					userId,
-					fileUploadParams.parentType,
-					fileUploadParams.parentId,
-					{ action: Actions.write, requiredPermissions: [Permission.FILESTORAGE_CREATE] }
-				);
+			it('should pass error', async () => {
+				const { uploadFromUrlParams, userId } = setup();
+
+				await expect(filesStorageUC.uploadFromUrl(userId, uploadFromUrlParams)).rejects.toThrowError();
 			});
+		});
 
-			it('should throw Error', async () => {
-				authorizationService.checkPermissionByReferences.mockRejectedValue(new ForbiddenException());
-				await expect(filesStorageUC.uploadFromUrl(userId, uploadFromUrlParams)).rejects.toThrow();
+		describe('WHEN uploadFile throws error', () => {
+			const setup = () => {
+				const { userId, uploadFromUrlParams, response } = getUploadFromUrlParams();
+				const error = new Error('test');
+
+				httpService.get.mockReturnValueOnce(of(response));
+				filesStorageService.uploadFile.mockRejectedValueOnce(error);
+
+				return { uploadFromUrlParams, userId };
+			};
+
+			it('should pass error', async () => {
+				const { uploadFromUrlParams, userId } = setup();
+
+				const expectedError = new NotFoundException(ErrorType.FILE_NOT_FOUND);
```

Comment:
This should fail... the mock return a js error. But you test a NotFoundException. 

.rejects.toThrow*Error*(expectedError); is maybe better.

Reply:
Test is correct. And toThrowError is an alias of toThrow